### PR TITLE
[tradfri] Update Californium to 2.0.0-RC1

### PR DESCRIPTION
**Pull request by flo-02-mu**
Signed-off-by: flo_02_mu <f.l.o.mueller@web.de>

Minor code changes to allow use the binding with the 2.0.0-M18 version of californium.

For discussion, see: https://community.openhab.org/t/tradfri-ignoring-unmatchable-piggy-backed-response-from-x-x-x-x-5684/79469/39

**Note:** This version won't build as long as openhab-core is not providing the same version of californium.

### CODE REVIEW

**Comment by cweitkamp** [comment]
FTR: I updated https://github.com/openhab/openhab-core/pull/1201 to use RC1 instead of M18.

File: bundles/org.openhab.binding.tradfri/pom.xml [hunk 1]
```diff
@@ -17,12 +17,12 @@
     <dependency>
       <groupId>org.eclipse.californium</groupId>
       <artifactId>californium-core</artifactId>
-      <version>1.0.7</version>
+      <version>2.0.0-M18</version>
```
Comment (cweitkamp) [inline bundles/org.openhab.binding.tradfri/pom.xml]:
```suggestion
      <version>2.0.0-RC1</version>
```

File: bundles/org.openhab.binding.tradfri/pom.xml [hunk 2]
```diff
@@ -17,12 +17,12 @@
...
     </dependency>
     <dependency>
       <groupId>org.eclipse.californium</groupId>
       <artifactId>scandium</artifactId>
-      <version>1.0.7</version>
+      <version>2.0.0-M18</version>
```
Comment (cweitkamp) [inline bundles/org.openhab.binding.tradfri/pom.xml]:
```suggestion
      <version>2.0.0-RC1</version>
```

File: bundles/org.openhab.binding.tradfri/src/main/java/org/openhab/binding/tradfri/internal/handler/TradfriGatewayHandler.java [hunk 3]
```diff
@@ -237,9 +240,13 @@ protected boolean obtainIdentityAndPreSharedKey() {
...
         } catch (JsonParseException e) {
-            logger.warn("Invalid response recieved from gateway '{}'", responseText, e);
+            logger.warn("Invalid response received from gateway '{}'", responseText, e);
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                    String.format("Invalid response recieved from gateway '%s'", responseText));
+                    String.format("Invalid response received from gateway '%s'", responseText));
+        } catch (ConnectorException |IOException e) {
+            logger.warn("Error connecting to gateway ",e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    String.format("Error connecting to gateway '%s'"));
```
Comment (cweitkamp) [inline bundles/org.openhab.binding.tradfri/src/main/java/org/openhab/binding/tradfri/internal/handler/TradfriGatewayHandler.java]: What should `%s` replaced with?

File: bundles/org.openhab.binding.tradfri/pom.xml (hunk 1 above)
Comment (flo-02-mu) [reply bundles/org.openhab.binding.tradfri/pom.xml]: Done.

File: bundles/org.openhab.binding.tradfri/pom.xml (hunk 2 above)
Comment (flo-02-mu) [reply bundles/org.openhab.binding.tradfri/pom.xml]: Done.

File: bundles/org.openhab.binding.tradfri/src/main/java/org/openhab/binding/tradfri/internal/handler/TradfriGatewayHandler.java (hunk 3 above)
Comment (flo-02-mu) [reply bundles/org.openhab.binding.tradfri/src/main/java/org/openhab/binding/tradfri/internal/handler/TradfriGatewayHandler.java]: Thanks for spotting this, I removed the placeholder.

File: bundles/org.openhab.binding.tradfri/src/main/java/org/openhab/binding/tradfri/internal/handler/TradfriGatewayHandler.java [hunk 4]
```diff
@@ -237,9 +240,13 @@ protected boolean obtainIdentityAndPreSharedKey() {
             logger.error("Illegal gateway URI '{}'", authUrl, e);
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
         } catch (JsonParseException e) {
-            logger.warn("Invalid response recieved from gateway '{}'", responseText, e);
+            logger.warn("Invalid response received from gateway '{}'", responseText, e);
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                    String.format("Invalid response recieved from gateway '%s'", responseText));
+                    String.format("Invalid response received from gateway '%s'", responseText));
+        } catch (ConnectorException |IOException e) {
+            logger.warn("Error connecting to gateway ",e);
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.tradfri/src/main/java/org/openhab/binding/tradfri/internal/handler/TradfriGatewayHandler.java]:
```suggestion
            logger.debug("Error connecting to gateway ",e);
```

File: bundles/org.openhab.binding.tradfri/pom.xml [hunk 5]
```diff
@@ -17,12 +17,12 @@
     <dependency>
       <groupId>org.eclipse.californium</groupId>
       <artifactId>californium-core</artifactId>
-      <version>1.0.7</version>
+      <version>2.0.0-RC1</version>
     </dependency>
     <dependency>
```
Comment (cweitkamp) [inline bundles/org.openhab.binding.tradfri/pom.xml]: Travis is failing right now. It complains about a missing dependency. You maybe have to add the `element-connector` to the POM as well.

Comment (flo-02-mu) [reply bundles/org.openhab.binding.tradfri/pom.xml]: That's interesting. It was not needed before and the code-change does not (directly) require it either. But anyway, now with the merged branch in core, it was easy to validate locally.

**Pull request merged**